Lib/glyphsLib/builder/builders.py

```diff
@@ -0,0 +1,259 @@
+                # in the font are appended after the listed glyphs, in the order
+                # in which they appear in the source file
+                glyph_order.append(glyph_name)
+
+            for layer in glyph.layers.values():
+                layer_id = layer.layerId
+                layer_name = layer.name
+
+                assoc_id = layer.associatedMasterId
+                if assoc_id != layer.layerId:
+                    if layer_name is not None:
+                        supplementary_layer_data.append(
+                            (assoc_id, glyph_name, layer_name, layer))
+                    continue
+
+                ufo = self._ufos[layer_id]
+                ufo_glyph = ufo.newGlyph(glyph_name)
+                self.to_ufo_glyph(ufo_glyph, layer, glyph)
+
+        for layer_id, glyph_name, layer_name, layer_data \
+                in supplementary_layer_data:
+            ufo_font = self._ufos[layer_id]
+            if layer_name not in ufo_font.layers:
+                ufo_layer = ufo_font.newLayer(layer_name)
+            else:
+                ufo_layer = ufo_font.layers[layer_name]
+            ufo_glyph = ufo_layer.newGlyph(glyph_name)
+            self.to_ufo_glyph(ufo_glyph, layer_data, layer_data.parent)
+
+        for ufo in self._ufos.values():
+            ufo.lib[glyphOrder_key] = glyph_order
+            if self.propagate_anchors:
+                self.to_ufo_propagate_font_anchors(ufo)
+            self.to_ufo_features(ufo)
+            self.to_ufo_kerning_groups(ufo, kerning_groups)
+
+        for master_id, kerning in self.font.kerning.items():
+            self.to_ufo_kerning(self._ufos[master_id], kerning)
+
+        return self._ufos.values()
+
+
+    @property
+    def instances(self):
+        """Get an iterator over interpolated UFOs of instances."""
+        # TODO?
+        return []
+
+
+    @property
+    def designspace(self):
+        """Get a designspace Document instance that links the masters together.
+        """
+        # TODO?
+        pass
+
+
+    @property
+    def instance_data(self):
+        instances = self.font.instances
+        if self._do_filter_instances_by_family:
+            instances = list(
+                filter_instances_by_family(instances,
+                                           self._instance_family_name))
+        instance_data = {'data': instances}
+
+        first_ufo = next(iter(self.master_ufos))
+
+        # the 'Variation Font Origin' is a font-wide custom parameter, thus it is
+        # shared by all the master ufos; here we just get it from the first one
+        varfont_origin_key = "Variation Font Origin"
+        varfont_origin = first_ufo.lib.get(GLYPHS_PREFIX + varfont_origin_key)
+        if varfont_origin:
+            instance_data[varfont_origin_key] = varfont_origin
+        return instance_data
+
+
+    # Implementation is spit into one file per feature
+    from .anchors import to_ufo_propagate_font_anchors, to_ufo_glyph_anchors
```

Comment:
I was also thinking.. perhaps instead of naming the methods `UFOBuilder.to_ufo_*`, and `GlyphsBuilder.to_glyphs_*`, which seems a bit redundant, we could replace the `to_ufo_*` and `to_glyphs_*` prefixes with a generic `build_*`, using the `as` keyword in the import statements.
E.g. inside `UFOBuilder` scope, you would do `from .anchors import to_ufo_glyph_anchors as build_glyph_anchors`, whereas inside `GlyphsBuilder`, `from .anchors import to_glyphs_glyph_anchors as build_glyph_anchors`.
WDYT?

Reply:
I don't mind the redundant naming because I think it makes things very clear inside the code: those names will only be used inside the implementations of the builders, with `self.` in front of it, and if you're looking at a file and stumble upon `self.build_anchors`, you will have to look around to find out in which direction the current function is going, whereas here it will always be very clear. Also having two names for the same function sounds confusing.

Reply:
ok, let's keep it like that